deploy/vagrant/bin/init_aws.py

```diff
@@ -17,26 +17,32 @@
 from util import info, warn, error
 
 
-def get_or_make_group(conn, name):
-    groups = conn.get_all_security_groups()
-    group = [g for g in groups if g.name == name]
-    if len(group) > 0:
-        return group[0]
+def get_or_make_group(conn, name, vpc=None):
+    if (vpc is None or vpc == ''):
+        groups = conn.get_all_security_groups([].append(name))
+    else:
+        groups = conn.get_all_security_groups([].append(name))
+        groups = [g for g in groups if (g.name == name and g.vpc_id == vpc)]
```

Comment:
Since `[].append(name)` is passed in, the `g.name == name` filter is redundant. Also, no matter what `vpc` is, the above always need to executed, redundant too.